### ISSUE TITLE
Web Workers

### ISSUE THREAD

**Issue by positlabs** [antimatter15/whammy#34]
Recording at high res / fps is pretty choppy. Is there some way we can improve this by offloading the `.toDataURL()` call to web workers? I assume that is where the bottleneck is, but I don't think you can reference dom elements in worker scripts.

Thoughts?

**Comment by antimatter15** [comment]
Yeah you can't reference DOM elements in worker scripts, so unfortunately
it probably isn't possible to parallelize by using webworkers.

**Comment by positlabs** [comment]
Ok, so we definitely need to call `canvas.toDataURL()` to get the webp format, right? What if we did something like cached the frames until `whammyVid.compile()` is called? `context.getImageData()` is probably pretty fast compared to `canvas.toDataURL()`. Then we could draw all of those frames back on to a canvas, and call `.toDataURL()` on them during the compile step. It might make compilation noticeably slower, but I think it's worth making things smoother during capture.

**Comment by positlabs** [comment]
I've just tried it. This method of delayed encoding works perfectly. It's almost an order of magnitude faster, during recording (at least for my application).

Current Whammy: https://youtu.be/QK9AOAvGMsw

Delayed encoding: https://youtu.be/99Kob5Xv0Yk

**Comment by rikkertkoppes** [comment]
delayed encoding turns out to be a huge benefit while recording video on mobile. Built in MediaRecorder api results in choppy video, Whammy saved the day here.

However, the encoding process really stalls the interface afterwards. 

I'd like to point to the OffscreenCanvas interface: 
- https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas
- https://www.chromestatus.com/feature/5424182347169792

This is not yet there, but may provide even further improvements during the compile phase by *actually* using webworkers for that
